src/globus_compute_common/messagepack/message_types/result.py

```diff
@@ -16,10 +16,23 @@ class ResultErrorDetails(pydantic.BaseModel):
     user_message: str
 
 
+@meta(message_type="result")
+class ResultV1(Message):
+    task_id: uuid.UUID
+    data: str
+    error_details: t.Optional[ResultErrorDetails]
+    task_statuses: t.Optional[t.List[TaskTransition]]
+
+    @property
+    def is_error(self) -> bool:
+        return self.error_details is not None
```

Comment:
Per verbal discussion, the driver for this class is mainly for testing and (future) developer assurance.  We've just landed on potentially moving this class to the test files directly, rather than exporting it in the production namespace.